src/Cmfd.cpp

```diff
@@ -1862,7 +1862,8 @@ void Cmfd::tallyCurrent(segment* curr_segment, FP_PRECISION* track_flux,
         int g = getCmfdGroup(e);
 
         for (int p=0; p < _num_polar; p++)
-          currents[g] += track_flux(p, e) * polar_weights[p] / 2.;
+          currents[g] += track_flux(p, e) * _quadrature->getWeight(azim_index,
```

Comment:
Why divide the weight by 2.0 here? Can't the `Quadrature` simply store the "correct" weight? Dividing by 2. is a non-negligible operation for each and every `segment` and polar angle.